src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationContext.tsx

```diff
@@ -0,0 +1,309 @@
+//     regionName: 'Belgium',
+//   },
+//   {
+//     isBeta: false,
+//     isProviderAvailable: true,
+//     isRegionAvailable: true,
+//     priority: 2,
+//     providerId: 'GCP',
+//     providerName: 'Google Cloud',
+//     regionId: 'us-central1',
+//     regionName: 'Iowa',
+//   },
+//   {
+//     isBeta: false,
+//     isProviderAvailable: true,
+//     isRegionAvailable: true,
+//     priority: 98,
+//     providerId: 'GCP',
+//     providerName: 'Google Cloud',
+//     regionId: 'us-central1-2',
+//     regionName: 'Iowa 2',
+//   },
+//   {
+//     isBeta: false,
+//     isProviderAvailable: true,
+//     isRegionAvailable: true,
+//     priority: 99,
+//     providerId: 'GCP',
+//     providerName: 'Google Cloud',
+//     regionId: 'us-west1',
+//     regionName: 'Oregon',
+//   },
+// ]
+// const FAKE_CLUSTERS: Cluster[] = [...FAKE_AWS, ...FAKE_AZURE, ...FAKE_GCP]
+
+export const CreateOrgProvider: FC<Props> = ({children}) => {
+  const [orgName, setOrgName] = useState(DEFAULT_CREATE_ORG_CONTEXT.orgName)
+  const [clusters, setClusters] = useState<ClusterMap>(
+    DEFAULT_CREATE_ORG_CONTEXT.clusters
+  )
+  const [currentProvider, setCurrentProvider] = useState<ProviderID>(
+    DEFAULT_CREATE_ORG_CONTEXT.currentProvider
+  )
+  const [currentRegion, setCurrentRegion] = useState<string>(
+    DEFAULT_CREATE_ORG_CONTEXT.currentRegion
+  )
+  const [createOrgLoadingStatus, setCreateOrgLoadingStatus] =
+    useState<RemoteDataState>(RemoteDataState.NotStarted)
+
+  const changeCurrentProvider = useCallback(
+    (provider: ProviderID) => {
+      setCurrentProvider(provider)
+    },
+    [setCurrentProvider]
+  )
+
+  const changeCurrentRegion = useCallback(
+    (region: string) => {
+      setCurrentRegion(region)
+    },
+    [setCurrentRegion]
+  )
+
+  const createOrg = useCallback((): Promise<void> => {
+    if (createOrgLoadingStatus === RemoteDataState.Loading) {
+      return new Promise(resolve => resolve(null))
+    }
+
+    setCreateOrgLoadingStatus(RemoteDataState.Loading)
+    return createNewOrg({
+      orgName,
+      provider: currentProvider,
+      region: currentRegion,
+    })
+      .then(() => {
+        setCreateOrgLoadingStatus(RemoteDataState.Done)
+      })
+      .catch(e => {
+        setCreateOrgLoadingStatus(RemoteDataState.Error)
+
+        throw e
+      })
+  }, [orgName, createOrgLoadingStatus, currentProvider, currentRegion])
+
+  const changeCreateOrgLoadingStatus = useCallback(
+    (status: RemoteDataState) => {
+      setCreateOrgLoadingStatus(status)
+    },
+    [setCreateOrgLoadingStatus]
+  )
+
+  useEffect(() => {
+    fetchClusterList().then(clusterData => {
+      // setClusters(r?.data)
+      // r
+      // let clusterData = r.data
+      // clusterData = FAKE_CLUSTERS
+
+      const clusterMap = clusterData.reduce((acc, curr) => {
+        if (!acc?.[curr.providerId]) {
+          acc[curr.providerId] = []
+        }
+        acc[curr.providerId].push(curr)
+
```

Comment:
I would change these to more descriptive variables so it's clear what `acc` and `curr` represent. Also, since we're just building a list of clusters, can't we just do this with a loop? I'm finding it difficult to understand what this is doing.